spec/controllers/saml_completion_controller_spec.rb

```diff
@@ -0,0 +1,35 @@
+require 'rails_helper'
+
+describe SamlCompletionController do
+  describe 'GET #index' do
+    render_views
+    include ActionView::Helpers::FormTagHelper
+
+    let(:form_action_regex) { /<form.+action=".+\/api\/saml\/authpost\d{4}.+"/ }
+    let(:saml_request) { 'abc123' }
+    let(:relay_state) { 'def456' }
+    let(:sig_alg) { 'aes256' }
+    let(:signature) { 'xyz789' }
+
+    context 'with SAML protocol params passed in appropriately via an internal redirect' do
+      let(:get_params) { "?SAMLRequest=#{saml_request}&RelayState=#{relay_state}&SigAlg=#{sig_alg}&Signature=#{signature}" }
+      let(:sp_session_request_url) { 'http://example.gov/api/saml/auth2022' }
+
+      before do
+        expect(controller).to receive(:sp_session).at_least(:once).and_return(
+          request_url: sp_session_request_url + get_params
```

Comment:
a few options here:

1. `Hash#to_query` for constructing the `get_params`
2. `UriService.add_params(sp_session_request_url, params_as_hash)`

Reply:
Updated here: https://github.com/18F/identity-idp/pull/6894/commits/53ad6badc09b0b1d464546fc0d436660ed6ef6d5#diff-449254b06f91b98dbac864d07cee959a85907e182bb3ea8b5143289abf017e85

Thanks, great suggestion!